### PR TITLE
Align protocol ids with the spec.

### LINKED ISSUE #9607: @pan-apple Remove service provisioniong, maybe multi-admin?
https://github.com/project-chip/connectedhomeip/blob/1657fcf16ee8be3041fd1a2ef30fcc30dbe0ed3d/src/app/server/Server.cpp#L171-L180

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 1657fcf16ee8be3041fd1a2ef30fcc30dbe0ed3d when #9200 was merged. cc @gjc13.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
* There are not ServiceProvisioning or OpCredentials protocols in the spec or
  in use in our code.
* Assign the right protocol id to InteractionModel.

Fixes https://github.com/project-chip/connectedhomeip/issues/9607

#### Problem
1. Implementation is not following spec.
2. Logs have protocol ids that don't match spec, which makes it hard to interpret them.

#### Change overview
Align protocol ids with the spec.

#### Testing
Compiles, passes CI (I hope).

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: src/protocols/Protocols.h [hunk 1]
```diff
@@ -74,12 +74,10 @@ class Id
...
 CHIP_STANDARD_PROTOCOL(SecureChannel, 0x0000)             // Secure Channel Protocol
-CHIP_STANDARD_PROTOCOL(Echo, 0x0001)                      // Echo Protocol
+CHIP_STANDARD_PROTOCOL(InteractionModel, 0x0001)          // Interaction Model Protocol
 CHIP_STANDARD_PROTOCOL(BDX, 0x0002)                       // Bulk Data Exchange Protocol
 CHIP_STANDARD_PROTOCOL(UserDirectedCommissioning, 0x0003) // User Directed Commissioning Protocol
-CHIP_STANDARD_PROTOCOL(InteractionModel, 0x0005)          // Interaction Model Protocol
-CHIP_STANDARD_PROTOCOL(ServiceProvisioning, 0x0007)       // Service Provisioning Protocol
-CHIP_STANDARD_PROTOCOL(OpCredentials, 0x0008)             // Operational Credentials
+CHIP_STANDARD_PROTOCOL(Echo, 0x0004)                      // Echo Protocol.  To be removed or standardized.
 CHIP_STANDARD_PROTOCOL(TempZCL, 0x0009) // For carrying ZCL-compatible messages, to be removed when we no longer use them
```
Comment (tcarmelveilleux) [inline src/protocols/Protocols.h:81]: What still uses this? Could we remove this now that IM is fully supported?

**Review by tcarmelveilleux — APPROVED** [review]

File: src/protocols/Protocols.h (hunk 1 above)
Comment (tcarmelveilleux) [reply src/protocols/Protocols.h:81]: This is actually getting removed in #9842

Comment (bzbarsky-apple) [reply src/protocols/Protocols.h:81]: It's still used by some Android code and the code being removed in #9842

**Review by LuDuda — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**